src/bufread.rs

```diff
@@ -110,12 +110,15 @@ impl<R: Read> Read for Cryptostream<R> {
         };
         if eof {
             self.finalized = true;
-            let write_bytes = self
-                .crypter
-                .finalize(&mut buf[bytes_written..])
-                .map_err(|e| Error::new(ErrorKind::Other, e))?;
-            // eprintln!("Wrote {} bytes to encrypted stream", write_bytes);
-            bytes_written += write_bytes;
+
+            if bytes_read > 0 {
```

Comment:
Actually, this isn't right. Imagine you have a cipher block size of 16 bytes and a full input source of 15 bytes. First time you read from it, `bytes_read == 15` and `!eof`, but the second time around, you'll have `eof` but with `bytes_read == 0` -- and you absolutely need to still call `finalize()` to get the last block padded and written.